{{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py

```diff
@@ -7,15 +7,18 @@
 Replace code below according to your needs.
 """
 from __future__ import annotations
-from typing import List
-from numpy.typing import ArrayLike
+from typing import TYPE_CHECKING, List, Any, Sequence, Tuple, Union
 
+if TYPE_CHECKING:
+    DataType = Union[Any, Sequence[Any]]
+    FullLayerData = Tuple[DataType, dict, str]
 
-def write_single_image(path: str, data: ArrayLike):
+
+def write_single_image(path: str, data: Any, meta: dict):
     """Writes a single image layer"""
```

Comment:
are we gonna do the rough "takes X and is expected to do Y and return Z" later?

Reply:
yes